### DISCUSSION TITLE
How add value changed logic when using @bind-Value?

### DISCUSSION THREAD

**Discussion by Strypper** [microsoft/fluentui-blazor#1126] [Q&A] [answered]
```html
<FluentDatePicker Label="Deadline" @bind-Value="@Content.Deadline" />
```

Like this example `ValueChanged` will duplicate with  @bind-Value

**Comment by dvoituron** [comment]
Hi, I don't quite understand your problem or question. Could you give me more details?

**Comment by Strypper** [comment]
If I use `@bind-Value` then I cannot use `ValueChanged` because of this error 
```bash
The component parameter 'ValueChanged' is used two or more times for this component. Parameters must be unique (case-insensitive). The component parameter 'ValueChanged' is generated by the '@bind-Value' directive attribute.		
```

**Comment by dvoituron** [comment, marked as answer]
Yes, this is the classic Blazor way for Two-way binding.

> Two-way binding in Blazor uses a naming convention. If we want to bind to a property named **SomeProperty**, then we need an event call-back named **SomeProperyChanged**. This call-back must be invoked any time the component updates SomeProperty.
> https://blazor-university.com/components/two-way-binding/

See also
- https://learn.microsoft.com/en-us/aspnet/core/blazor/components/data-binding

**Comment by KurimasuTanaka** [comment]
It's also possible to bind Switch value to a parameter and write additional logic in setter. 
```
<Switch Label="Value" @bind-Value="bindedValue" />

@code {
    private bool _bindedValue;
    [Parameter]
    public bool bindedValue {
        get => _bindedValue;
        set 
        {
             //your custom logic
        }
    }
```

**Reply by vnbaaij** [reply]
For .NET 7 and higher you can use `@bind-...:after` to specify a method that is called after the binding takes place. So, for example:
```
@bind-Value="@SearchTerm" @bind-Value:after="@StartNewSearchAsync"
```

**Reply by woaidianqian** [reply]
I would like to add something

```
                                          
 <FluentSwitch @bind-Value="proposalSwitchValue" @bind-Value:after="async () => {await OnProposalSwitchChanged(proposalSwitchValue); }" CheckedMessage="On" UncheckedMessage="Off" >
                                                

@bind-Value="proposalSwitchValue" @bind-Value:after="async () => {await OnProposalSwitchChanged(proposalSwitchValue); }``
```
